src/main/java/org/craftercms/studio/impl/v1/content/pipeline/FormDmContentProcessor.java

```diff
@@ -215,8 +220,23 @@ protected ContentItemTO createNewFile(String site, ContentItemTO parentItem, Str
                 } else {
                     properties.put(ItemMetadata.PROP_LOCK_OWNER, user);
                 }
-                objectMetadataManager.setObjectMetadata(site, parentItem.getUri() + FILE_SEPARATOR + fileName, properties);
-                result.setCommitId(objectMetadataManager.getProperties(site, parentItem.getUri() + FILE_SEPARATOR + fileName).getCommitId());
+                objectMetadataManager.setObjectMetadata(site, parentItem.getUri() + FILE_SEPARATOR + fileName,
+                        properties);
+                result.setCommitId(objectMetadataManager.getProperties(site, parentItem.getUri() + FILE_SEPARATOR +
+                        fileName).getCommitId());
+
+                // Item
+                // TODO: get local code with API 2
+                Item item = itemServiceInternal.instantiateItemAfterWrite(site,
```

Comment:
Please add `lock` as a parameter (optional?) such that the locking code is not repeated.